src/chocolatey.console/Program.cs

```diff
@@ -238,20 +238,45 @@ private static void add_assembly_resolver()
 #else
                 var chocolateyPublicKey = ApplicationParameters.UnofficialChocolateyPublicKey;
 #endif
+                if (!requestedAssembly.get_public_key_token().is_equal_to(chocolateyPublicKey))
+                {
+                    // This resolver is only loading official extensions for Chocolatey and ILMerged assemblies.
+                    // Everything else is not handled by this resolver.
+                    return null;

```

Comment:
It's not this line in particular, but the return is confusing when looking at the diff (see screenshot below). As the diff shows we're inside of void method `add_assembly_resolver()` and yet we are returning things. I had to expand the view to see that we are in a delegate definition. Would it be an idea to move this delegate definition into its own method and pass the method reference to the delegate handler?

![2022-11-29_09-16-02](https://user-images.githubusercontent.com/30301021/204598137-cfe027a6-6173-4566-8bcc-d62b73a79f12.png)

I'm not sure if such a refactor necessarily belongs in a bugfix release though, so perhaps if we decide we want to refactor like that we can open an issue and address it in a future release.

Reply:
Yeah I'm all for cleaning this up some more. Unsure if it's something we want to do at this point, but happy to discuss. It's a relatively simple change to make, just moving some code around a little more.

/cc @admiringworm

Reply:
With the current codebase, I am fine with short-circuiting if the requested assembly isn't officially signed.
But do note that this will require more changes needed when/if we ever decide in the future to open up (or partially open) creating extensions by non-officially signed binaries.